flink-runtime/src/main/java/org/apache/flink/runtime/iterative/task/AbstractIterativeTask.java

```diff
@@ -125,6 +125,16 @@ protected void initialize() throws Exception {
 	}
 
 	@Override
+	protected void postRun() throws Exception {
```

Comment:
The instanceof for ResettableDriver like
`if (this.driver instanceof ResettableDriver) {`

was being done in AbstractITerativetask. So I thought if we could clearly seperate out what the finalize clause after every run() can do should be differentiated based on the current instance executing it. Let BatchTask do only driver.cleanup and let the AbstractIterativeTask do based on the type of the driver. I can remove it if not needed. That was just for code clarity.